metpy/calc/indices.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) 2008-2015 MetPy Developers.
+# Distributed under the terms of the BSD 3-Clause License.
+# SPDX-License-Identifier: BSD-3-Clause
+"""Contains calculation of various derived indicies."""
+import numpy as np
+
+from . import mixing_ratio, saturation_vapor_pressure
+from .tools import get_layer
+from ..constants import g, rho_l
+from ..package_tools import Exporter
+from ..units import units
+
+exporter = Exporter(globals())
+
+
+def precipitable_water(dewpt, p, top=400 * units('hPa')):
+    r"""Calculate precipitable water through the depth of a sounding.
+
+    Default layer depth is sfc-400 hPa. Formula used is:
+
+    \begin{align}
+    \frac{1}{pg} \int\limits_0^d x \,dp
+    \end{align}
+
+    from [Tsonis2008]_, p. 170.
+
+    Parameters
+    ----------
+    dewpt : array-like
+        Atmospheric dewpoint profile
+    p : array-like
+        Atmospheric pressure profile
+    top: `pint.Quantity`
+        The top of the layer, specified in pressure.
+
+    Returns
+    -------
+    `pint.Quantity'
+        The precipitable water in the layer, in inches
+
+    """
+    pw_layer, dewpt_layer = get_layer(p, dewpt, depth=p[0] - top)
+
+    w = mixing_ratio(saturation_vapor_pressure(dewpt_layer), pw_layer)
+
+    sort_inds = np.argsort(pw_layer)
+    pw_layer = pw_layer[sort_inds]
+    w = w[sort_inds]
+
+    pw = np.trapz(w, pw_layer) / (g * rho_l)
+    pw = ((pw.magnitude * 100) * units('meter')).to('inches')
```

Comment:
Since pressure is in hPa, the units from the np.trapz function come out to be this hilarious mess of  kilograms / (hectometers * seconds ^2), which then combines with the almost-as-messy units on the (g * rho_l) term to get meters. However, units seems to be having some trouble working with hectometers, and results in some really wrong answers. Is there a better way to do this? Also, to be useful, I think the output from the function should probably be in inches or millimeters, since precipitable water plotted in straight meters is probably not the best idea. For the moment I've kept the output in inches and manually converted from hectometers to meters.